### PR TITLE
web render update - #1116

### PR DESCRIPTION
**Pull request by proddy**
lots of minor refactoring to improve performance in the Web. There was a lot of re-rendering when inputting data into a modal dialog box. See https://github.com/emsesp/EMS-ESP32/issues/1116. 
The big change is making separate react component classes for the dialog modals, preventing excessive rendering of the parent classes. it should improve performance.

Note:
It does break backward compatibility with 3.6.0-dev-9 with Entity Customization as the payload format has changed. The device id and type id are now numbers (not strings). And the JSON and endpoints are called `entities` to keep with the standard naming convention.

### CODE REVIEW

**Comment by MichaelDvP** [comment]
Wow, 120 files changed! Thats a bit to much to review all the code. I've only tested.
The rendering is really fast now. Only things i've seen:
- system log now have a update button, but this button is always there. Better use dirty flags and show it as `Apply changes()`.
Another thing, old, but maybe need some rework: the manage user page shows the update button outlined and always, after changing a user-password you have to press the update button. Also here the `Apply changes()` button on demand will be more clear.

**Comment by proddy** [comment]
> Wow, 120 files changed! Thats a bit to much to review all the code. I've only tested. The rendering is really fast now. Only things i've seen:
> 
> * system log now have a update button, but this button is always there. Better use dirty flags and show it as `Apply changes()`.
> 
> Another thing, old, but maybe need some rework: the manage user page shows the update button outlined and always, after changing a user-password you have to press the update button. Also here the `Apply changes()` button on demand will be more clear.

yes, I think every single file was touched because of new ESLint rules, one of which re-orders the imports via `yarn run lint-fixall` does that. The main changes are in the Settings*.tsx and the way the Dialogs work. So I wouldn't bother doing a code review, maybe check that the Custom Entities still works as I changed the type of device_id and type_id (they were strings first to accept hex numbers in the TextField input. 

I'll work on the DashboardData next as this is also really inefficient when you have many entities showing on the screen.

Good call on the System Log Update button and also adding the Apply Updates to the User management screens, thanks.

File: interface/src/project/SettingsEntitiesDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+              margin="normal"
+              fullWidth
+              type="number"
+              value={editItem.offset}
+              onChange={updateFormValue}
+            />
+          </Grid>
+          <Grid item xs={4}>
+            <ValidatedTextField
+              name="val_type"
```
Comment (MichaelDvP) [inline interface/src/project/SettingsEntitiesDialog.tsx:160]: name should be also `value_type`

**Review by MichaelDvP — APPROVED** [review]
Works good. I've changed a lot of settings, but maybe not all.

File: interface/src/project/SettingsEntitiesDialog.tsx (hunk 1 above)
Comment (proddy) [reply interface/src/project/SettingsEntitiesDialog.tsx:160]: fixed

File: interface/src/project/SettingsEntities.tsx [hunk 2]
```diff
@@ -129,147 +108,135 @@ const SettingsEntities: FC = () => {
...
     }
-    setEntityItem(undefined);
+  };
+
+  const addEntityItem = () => {
+    setCreating(true);
+    setSelectedEntityItem({
+      id: Math.floor(Math.random() * (Math.floor(200) - 100) + 100),
+      name: '',
+      device_id: 11,
```
[the remark this answers is not in the document]
Comment (proddy) [reply interface/src/project/SettingsEntities.tsx:190]: fixed

**Comment by MichaelDvP** [comment]
The update button in weblog does not work proper. I get a Offset is out of bounds error. Also the blocker is not shown, navigation is blocked while the update button shows, but the banner is missing.

**Comment by proddy** [comment]
Ok. I'll fix.

**Comment by proddy** [comment]
> Ok. I'll fix.

fixed.

**Comment by proddy** [comment]
Think it's mostly done now. Now I need to test it.

**Comment by proddy** [comment]
@MichaelDvP I think its working ok. Let me know what you think. Noticeable changes are speed in the web pages, Sensors have moved to a new tab and I made the Device Data scrollable.

**Comment by MichaelDvP** [comment]
Some first impressions:
- Speed is fine. After updating packages with react-router 6.11 routing is broken, i have to go back to 6.10.
- Renaming dallas to temperaturesensors breaks mqtt and api (including @tp1de iobroker adapter). Not sure if this is really worth it.
- Sensors in the new tab is ok (i like the old style on one page more, maybe with some time...)
- entities in the scrollbox is not good, with some more devices the box is completly vanished, or only half a line in height:
Screenshot 2023-05-01 at 13-09-07 EMS-ESP
- Remove the `<Grid> from `edit dialog: 
- Screenshot 2023-05-01 at 13-22-03 EMS-ESP
- Edit dialog is only string input, number input with up/down arrows is missing (step should be string again, float is not working properly)
- Update button in weblog: when saving there should be a `setOrigData(data)`to allow changing setting back. 

to be continued...

**Comment by proddy** [comment]
this PR is getting big so I'm planning to push the changes in a few days, which means different UI and the rename of dallas.

**Pull request merged**